From 0cd50aecf2774c2fd542a07071f171943e3a4da5 Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Sun, 2 Feb 2020 12:32:59 +0200
Subject: [PATCH 14/22] cp(1): Install signal handler only if SIGINFO is
 available.

---
 bin/cp/cp.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/bin/cp/cp.c b/bin/cp/cp.c
index a8f9ba64e7..ae299cf08f 100644
--- a/bin/cp/cp.c
+++ b/bin/cp/cp.c
@@ -77,7 +77,9 @@ enum op { FILE_TO_FILE, FILE_TO_DIR, DIR_TO_DNE };
 
 static int copy(char *[], enum op, int);
 static int mastercmp (const FTSENT * const *, const FTSENT * const *);
+#ifdef SIGINFO
 static void siginfo (int);
+#endif
 
 int
 main(int argc, char *argv[])
@@ -165,7 +167,9 @@ main(int argc, char *argv[])
 		fts_options &= ~FTS_PHYSICAL;
 		fts_options |= FTS_LOGICAL | FTS_COMFOLLOW;
 	}
+#ifdef SIGINFO
 	signal(SIGINFO, siginfo);
+#endif
 
 	/* Save the target base in "to". */
 	target = argv[--argc];
@@ -499,9 +503,10 @@ mastercmp(const FTSENT * const *a, const FTSENT * const *b)
 	return (0);
 }
 
+#ifdef SIGINFO
 static void
 siginfo(int sig __unused)
 {
-
 	info = 1;
 }
+#endif
-- 
2.25.0

