From 41c6a0f3b4aef5bce5b0c4837592ee8d1f4e3b69 Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Wed, 9 Oct 2019 11:34:28 +0300
Subject: [PATCH 1/7] OpenSSH: Update README.

---
 crypto/openssh/README.DRAGONFLY | 1 -
 1 file changed, 1 deletion(-)

diff --git a/crypto/openssh/README.DRAGONFLY b/crypto/openssh/README.DRAGONFLY
index 9d08209398..d75203b954 100644
--- a/crypto/openssh/README.DRAGONFLY
+++ b/crypto/openssh/README.DRAGONFLY
@@ -32,7 +32,6 @@ Update config.h and openbsd-compat in lib/libssh.
 
 Used in:
 lib/libssh/
-lib/libpam/modules/pam_ssh/
 libexec/sftp-server/
 libexec/ssh-keysign/
 libexec/ssh-pkcs11-helper/
-- 
2.23.0

