diff --git a/lib/libthread_xu/thread/thr_mutex.c b/lib/libthread_xu/thread/thr_mutex.c
index 3ed503c265..687ec160b4 100644
--- a/lib/libthread_xu/thread/thr_mutex.c
+++ b/lib/libthread_xu/thread/thr_mutex.c
@@ -474,6 +474,8 @@ _pthread_mutex_lock(pthread_mutex_t *m)
 	struct pthread *curthread;
 	int	ret;
 
+	_thr_check_init();
+
 	if (__predict_false(m == NULL))
 		return(EINVAL);
 
@@ -497,6 +499,8 @@ __pthread_mutex_timedlock(pthread_mutex_t * __restrict m,
 	struct pthread *curthread;
 	int	ret;
 
+	_thr_check_init();
+
 	if (__predict_false(m == NULL))
 		return(EINVAL);
 
diff --git a/lib/libthread_xu/thread/thr_once.c b/lib/libthread_xu/thread/thr_once.c
index ef172f3f0a..e970c49489 100644
--- a/lib/libthread_xu/thread/thr_once.c
+++ b/lib/libthread_xu/thread/thr_once.c
@@ -61,6 +61,8 @@ _pthread_once(pthread_once_t *once_control, void (*init_routine) (void))
 {
 	int wakeup = 0;
 
+	_thr_check_init();
+
 	if (once_control->state == ONCE_DONE)
 		return (0);
 	_pthread_mutex_lock(&once_lock);
